core/domain/collection_domain.py

```diff
@@ -230,13 +263,54 @@ def create_default_node(cls, exploration_id):
         return cls(exploration_id)
 
 
+class CollectionDict(TypedDict, total=False):
+    """Type for the collection object"""
+
+    id: str
+    title: str
+    category: str
+    objective: str
+    language_code: str
+    tags: List[str]
+    schema_version: int
+    nodes: list[CollectionNodeDict]
+    version: int
+    created_on: Union[datetime.datetime, str, None]
+    last_updated: Union[datetime.datetime, str, None]
+
+
+class CollectionVersionDict(CollectionDict, total=False):
+    """Type for the collection Dict based on different versions"""
+
+    skills: Dict[str, Dict[str, Any]]
```

Comment:
Why `Any`?

Reply:
Now, it is fixed. skills is dict with following format :
```
skill_id -> string : {
                'name': skill_name -> string,
                'question_ids': [] -> list of strings
            }
```